grammar.y

```diff
@@ -972,7 +986,35 @@ create_table_stmt:
     if len($5) > MaxAllowedColumns {
       yylex.(*Lexer).AddError(&ErrTooManyColumns{ColumnCount: len($5), MaxAllowed: MaxAllowedColumns})
     }
-    $$ = &CreateTable{Table: $3, Columns: $5, Constraints: $6}
+
+    // We have to replace a primary key table constraint with an equivalent column constraint primary key,
+    // so we can add the autoincrement flag, as part of the rules of the Tableland Protocol.
+    // 
+    // That happens because a primary key table constraint that references a single INTEGER column
+    // would be an alias to rowid. For cases where a column becomes an alias to rowid we want to force the AUTOINCREMENT.
+    // 
+    // The exception to the above rule is when a table constraint primary key has order DESC. In that case, we replace with an 
+    // equivalent column constrain without forcing AUTOINCREMENT.
+    var tableConstraintPK *TableConstraintPrimaryKey
+    var index int
+    for i, tableConstraint := range $6 {
+      if tcpk, ok := tableConstraint.(*TableConstraintPrimaryKey); ok && len(tcpk.Columns) == 1 {
+        tableConstraintPK = tcpk
+        index = i
+      }
+    }
+
+    for _, columnDef := range $5 {
```

Comment:
search for an `INTEGER` column that has no primary key and has the same name as the `tableConstraintPK` column name.

If that's the case, remove the table constraint and add the column constraint

The autoincrement flag depends on the order of the `tableConstraintPK`